### PR TITLE
8269087: CheckSegmentedCodeCache test fails in an emulated-client VM

### PR DESCRIPTION
**Pull request by mychris**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8269087](https://bugs.openjdk.java.net/browse/JDK-8269087): CheckSegmentedCodeCache test fails in an emulated-client VM

### Reviewers
 * [Igor Veresov](https://openjdk.java.net/census#iveresov) (@veresov - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4540/head:pull/4540` \
`$ git checkout pull/4540`

Update a local copy of the PR: \
`$ git checkout pull/4540` \
`$ git pull https://git.openjdk.java.net/jdk pull/4540/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4540`

View PR using the GUI difftool: \
`$ git pr show -t 4540`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4540.diff">https://git.openjdk.java.net/jdk/pull/4540.diff</a>

</details>

### CODE REVIEW

**Review by veresov — COMMENTED** [review]
Under what circumstances does the VM print "-XX:+TieredCompilation not supported in this VM" ? Is it still really possible?

**Comment by mychris** [comment]
After thinking a bit more about this, I guess it is irrelevant if the VM does still print the warning or not. It might if neither C1, nor C2 is compiled into it, so maybe in a zero VM?

But if I understand the test correctly, the reason for checking for this error message is, that certain checks should only be done if C2 is present, so I guess the newly added check should be enough to catch all cases.

But maybe I am misunderstanding the test, because if that is the case, then why isn't the test simply tagged as requires c2?

**Comment by mychris** [comment]
I changed the test to use the `Platform` class to check if tiered compilation is supported or not. The test case should now behave the same as it has before, but the check is more robust and works in an emulated-client as well.

File: test/hotspot/jtreg/compiler/codecache/CheckSegmentedCodeCache.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ private static void verifySegmentedCodeCache(ProcessBuilder pb, boolean enabled)
                 out.shouldContain(NON_METHOD);
             } catch (RuntimeException e) {
                 // Check if TieredCompilation is disabled (in a client VM)
-                if (!out.getOutput().contains("-XX:+TieredCompilation not supported in this VM")) {
+                if (Platform.isTieredSupported()) {
```
Comment (veresov) [inline test/hotspot/jtreg/compiler/codecache/CheckSegmentedCodeCache.java:61]: That's probably not a complete set of conditions of when the segmented code cache is disable. But if it solves the current issue, I think that's fine.

**Review by veresov — APPROVED** [review]
I ran this through internal testing. All clean.

**Comment by mychris** [comment]
Thanks for the review, Igor. Is this trivial enough or do I need a second reviewer?

**Review by vnkozlov — APPROVED** [review]

**Comment by mychris** [comment]
Thanks for the review.
/integrate

**Pull request closed without merging**